### DISCUSSION TITLE
HASPone connected to MQTT but core blueprint not working

### DISCUSSION THREAD

**Discussion by dexterprinceton** [HASwitchPlate/HASPone#244] [Troubleshooting] [answered]
Hey,
I am running HA Core 2024.05.2 and my HASP one says its connected to MQTT. I can control the page displayed and LCD screen via MQTT device in integrations screen of HA, however, the CORE blueprint and any other blue print I add does not appear to work or display any changes I am making...?  Any idea what would be causing this behavior?

**Comment by aderusha** [comment, marked as answer]
There has been a breaking change in HA which requires a firmware update.  I've been cooking 1.06 for test for a while now and am getting ready to release (weather forecast blueprint is the current holdup due to yet another breaking change, HA devs love breaking things).

You can test that out by downloading this and flash it to the device: https://raw.githubusercontent.com/HASwitchPlate/HASPone/dev/Arduino_Sketch/HASwitchPlate.ino.d1_mini.bin

It may require running through this to get things to work:  https://github.com/HASwitchPlate/HASPone/wiki/Fix-common-HASPone-issues

**Reply by dexterprinceton** [reply]
Hey dude,
I figured out the weather forecast thing by creating a sensor that provides the data and using a separate blueprint to display the sensor data.

Your firmware update worked a treat and I wanted to take this opportunity to say I really appreciate your support.  I have three of these up and running for various applications and they were a joy to build and program.  Its also great to see you still supporting this device years down the road.

Cheers!

**Comment by aderusha** [comment]
Hey thanks for the kind words!  I have these things all over my house and use them constantly so I'll probably have to keep updating stuff just because I love `em :D  

I've worked out a new solution for the weather forecast and the blueprints are also in the dev branch while I get a release together.  Check out [this](https://github.com/HASwitchPlate/HASPone/blob/dev/Home_Assistant/blueprints/hasp_Display_Weather_Forecast.yaml) and [this](https://github.com/HASwitchPlate/HASPone/blob/dev/Home_Assistant/blueprints/hasp_Display_Weather_Forecast_High_Low.yaml)
